doc/Consensus.md

```diff
@@ -0,0 +1,79 @@
+# Consensus
+
+## Data Formats
+
+
+## Rules
+
+#### Output Uniqueness
+
+* Output commitments in the UTXO set must be unique at all times.
+* A block cannot be included when it contains an output commitment which already exists in the UTXO set.
+  * Block is still invalid even if it also contains an input with said commitment.
+* A block cannot contain 2 outputs with the same commitment.
+  * Block is still invalid even if it also contains an input with said commitment.
+* A block *can* contain an input which spends an output created in the same block, provided the commitment was not already in the UTXO set.
+  * To better support payment proofs, the outputs should still be added to the TXO MMR, but should be marked as spent.
+---
+
+#### Ordering
+
+* Inputs in a block must be in ascending order by their raw commitment value.
+* Outputs in a block must be in ascending order by their raw commitment value.
+* Kernels in a block must be in ascending order by their raw hashed value.
+* Signed owner messages must be in ascending order by their raw hashed value.
+---
+
+#### PMMRs
+
+* After adding all kernels from a block to the end of the kernel MMR, the root must match the header's kernel root.
+* After adding all kernels from a block to the end of the kernel MMR, the size must match the header's kernel size.
+* After adding all outputs from a block to the end of the output MMR, the root must match the header's output root.
+* After adding all outputs from a block to the end of the output MMR, the size must match the header's output size.
+* After adding all rangeproofs from a block to the end of the rangeproof MMR, the root must match the header's rangeproof root.
+* After adding all rangeproofs from a block to the end of the rangeproof MMR, the size must match the header's output size.
+---
+
+#### UTXO LeafSet
+
+* TODO
+---
+
+#### Signatures
+
+* Each kernel must have a valid signature of the `signature_message` for the kernel's `commitment`.
+  * TODO: Define`signature_message` serialization for each kernel type
```

Comment:
Space after "Define"

Reply:
That's intentional:
![image](https://user-images.githubusercontent.com/24623306/103761802-42b20480-4fe5-11eb-9fe4-796aeeebc2aa.png)